### ISSUE TITLE
Tags are not replaced, but getting appended

### ISSUE THREAD

**Issue by rrjanbiah** [solidjs/solid-meta#8]
Say, for example, if `index.html` is having `<title>` tag already, the new `title` tag is getting appended.. so, browser is taking the first one (which is used in the `index.html`).. in turn looks like the module is not working. 

As a quick fix, I had to remove the `<title>` tag from the `index.html`. But, that's not the correct solution.. as I want _somewhat_ SEO friendly without SSR for the static hosting.

Apologies for my poor English and hope the issue is clear to the author.

**Comment by ryansolid** [comment]
We can't diff what is outside of our control. SSR does a trick where it replaces by writing an attribute to server generated meta tags. We should document this. But if you add a `data-sm=""` attribute to any of your meta tags they will be removed when the client hydrates and replaced with what the library generates.

**Comment by rrjanbiah** [comment]
@ryansolid Thanks for sharing the hidden treasure :-) `<title data-sm="">My title</title>` worked great to avoid the appending. Feel free to close the issue unless you want to track the documentation part.

**Comment by andreas-ateles** [comment]
When setting title why not just remove any existing `<title>` tags? As there should never be more than one of those.

**Comment by rrjanbiah** [comment]
@andreas-ateles As I mentioned in original message, I need it for static hosting. For example, think about it when using Facebook debugger or Twitter cards. With it, it would also resolve the root/default page's `title` for all the pages.

**Comment by outbackStack** [comment]
It appends the head as well if you use under a route like bellow - client side.


```js

import { Routes, Route } from "solid-app-router"
const Home = lazy(() => import("./pages/Home"));

export default function App() {
  return (<>
    <h1>My Site with Lots of Pages</h1>
    <Routes>
      <Route path="/" element={<Home/>} />
      <Route path="/about"  element={<Home/>}  />
    </Routes>
  <>)
}
```

Result:
<img width="664" alt="Screen Shot 2022-06-21 at 11 31 59 am" src="https://user-images.githubusercontent.com/16639496/174697820-839027fc-5bdc-4118-9a5f-38583840478c.png">

**Comment by ryansolid** [comment]
You have multiple MetaProviders. Each acts independently. Wrap the whole application in a single MetaProvider and it should work fine.

See: https://codesandbox.io/s/solid-meta-reproduction-3r1qgb

**Comment by outbackStack** [comment]
thanks @ryansolid, it's working now...
Loving solid-js so far 😀

**Comment by nathan-alden-sr** [comment]
Am I missing something? My `index.html` has `<title data-sm="">...</title>`. My app only has a single `<MetaProvider>` at the top-level. Then, I provide a `<Title>` but it's still appending to the markup.

```html
<title data-sm>Hardcoded Title</title>
...
<title>Set by Title component</title>
```

**Comment by ryansolid** [comment]
Yeah I thought this might be the case. We used to do something simple and just remove all the existing tags but now we try to upgrade but if the id doesn't match it stays I imagine. We probably need a convention here. I'm going to ask @nksaraf for his opinion as well.

**Comment by nathan-alden-sr** [comment]
See https://github.com/solidjs/solid-meta/issues/17 for what I think is an improved design that can help mitigate these issues but also provide more flexibility.

**Comment by AlexTMjugador** [comment]
For client-side rendering, I worked around this behavior, which was undesirable for me due to the reasons stated above, by assigning an ID to the `<title>` element in the `index.html` and removing it on my `index.tsx` file. It's not a pretty solution, and there will be a tiny, unnoticeable time window in which the page won't have a title at all, but it is more SEO-friendly than the alternative.

**Comment by rrjanbiah** [comment]
@AlexTMjugador The term SEO usually refers to the SSR or what HTML content is presented to the crawler bots. So, your JavaScript based solution may not be ideal here
